include/networkit/graph/Graph.hpp

```diff
@@ -590,25 +592,29 @@ class Graph final {
      */
     template <bool InEdges = false> class NeighborRange {
       public:
-        NeighborRange(const Graph &G, node u) : G(G), u(u){};
+        NeighborRange(const Graph &G, node u) : G(&G), u(u){};
+
+        NeighborRange() : G(nullptr) {};
 
         NeighborIterator begin() const {
             if (InEdges)
-                return NeighborIterator(G.inEdges[u].begin());
-            return NeighborIterator(G.outEdges[u].begin());
+                return NeighborIterator(G->inEdges[u].begin());
+            return NeighborIterator(G->outEdges[u].begin());
         }
 
         NeighborIterator end() const {
             if (InEdges)
-                return NeighborIterator(G.inEdges[u].end());
-            return NeighborIterator(G.outEdges[u].end());
+                return NeighborIterator(G->inEdges[u].end());
+            return NeighborIterator(G->outEdges[u].end());
         }
 
       private:
-        const Graph &G;
-        const node u;
+        const Graph *G;
+        node u;
     };
 
+	typedef NeighborRange<false> UndirectedNeighborRange;
```

Comment:
While it's not in our style guide, I feel `typedefs` are superseded by `using. So please rewrite  `using UndirectedNeighborRange = NeighborRange<false>`. While you're at it: could you please also refactor the typedefs in the iterators themselves?

Reply:
On a more general node: This iterator is also used to iterate over directed out neighbors. So the name is somewhat misleading. In anycase, please also adapt the signature of `Graph::neighborRange`.